### DISCUSSION TITLE
Image is not lazy loaded when having lazy in img loading attribute

### DISCUSSION THREAD

**Discussion by zluo01** [tauri-apps/tauri#9825] [Q&A]
Hi team,
Tt seems to me that tauri would not honor img `loading="lazy"` attribute to lazy load image on page load. I have test this on Linux and notice that in development mode, on page load, it will have network calls to all images instead of those current in view only.

Seems to me this feature is already supported in all major browsers as mentioned in [here](https://web.dev/articles/browser-level-image-lazy-loading), thus I assume this is a behavior happens on rendering layer and since tauri use its own rendering library this is not support ? 

Thus, I wonder if there is any plan on supporting this if possbile ? Or is there any suggestion on achieving this ? I aware about third party libraries or frameworks that support lazy load image but wondering if there is anyway achieving this without the need for extra dependencies ?

Thanks.

**Comment by FabianLars** [comment]
> thus I assume this is a behavior happens on rendering layer and since tauri use its own rendering library this is not support ?

Tauri does not have its own rendering layer/library. This is all up to the browser engine that we have no control over. In browser terms, support for it in webkit based browser (in tauri = macos and linux) is fairly recent: https://caniuse.com/loading-lazy-attr so i'd imagine that either the webkit version you're testing this on is too old, or it's similar to `async` and `defer` and broken in webkitgtk (not 100% ruled out that we're doing something wrong in loading the assets but those keyworks work fine outside of linux so idk)

**Reply by zluo01** [reply]
I try on mac but seems like it is also not lazy load as well. As the examples shown below, ideally I would expect only first 6 images are loaded since rest of the components are not in the view yet. However, from the console, I do see all 21 image being loaded with total 4.21 mb. I make sure I update to the latest mac os version.

<img width="1920" alt="Screenshot 2024-05-26 at 20 16 27" src="https://github.com/tauri-apps/tauri/assets/12102011/0140b991-7a22-4647-85cb-c400aa7ae44d">

<img width="1920" alt="Screenshot 2024-05-26 at 20 23 15" src="https://github.com/tauri-apps/tauri/assets/12102011/df556cc1-f20c-4933-9d1a-56b123b24447">


```
[✔] Environment
    - OS: Mac OS 14.5.0 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.78.0 (9b00956e5 2024-04-29) (Homebrew)
    ✔ cargo: 1.78.0
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: 
    - node: 21.7.3
    - pnpm: 9.1.2
    - npm: 10.5.0

[-] Packages
    - tauri [RUST]: 1.6.7
    - tauri-build [RUST]: 1.5.2
    - wry [RUST]: 0.24.10
    - tao [RUST]: 0.16.9
    - @tauri-apps/api [NPM]: 1.5.6
    - @tauri-apps/cli [NPM]: 1.5.14

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: React
    - bundler: Vite

```

**Comment by FabianLars** [comment]
discussion moved to https://github.com/tauri-apps/tauri/issues/10120
